docs/fundamentals/write-operations.txt

```diff
@@ -0,0 +1,489 @@
+--------------------------------------
+
+An **upsert** operation lets you perform an update or insert in a single
+operation. This operation streamlines the task of updating a document or
+inserting one if it does not exist.
+
+To specify an upsert in an ``update()`` method, set the ``upsert`` option to
+``true`` as shown in the following code example:
+
+.. code-block:: php
+   :emphasize-lines: 4
+   :copyable: false
+
+   YourModel::where(<match criteria>)
+      ->update(
+          [<update data>],
+          [upsert => true]);
+
+When the ``update()`` method is chained to a query, it performs one of the
+following actions:
+
+- If the query matches documents, the ``update()`` method  updates the matching
+  documents.
+- If the query matches zero documents, the ``update()`` method inserts a
+  document that contains the update data and the equality match criteria data.
+
+Upsert Example
+~~~~~~~~~~~~~~
+
+This example shows how to pass the ``upsert`` option to the  ``update()``
+method to perform an update or insert in a single operation. It also includes
+example results in which it performs an insert.
+
+The following example shows how to perform an update or insert in a single
+operation by passing the ``upsert`` option to the ``update()`` method. The
+example output shows the result if the match criteria passed to the
+``where()`` method matches zero documents in the collection. Click the
+:guilabel:`VIEW OUTPUT` button to see the document inserted by running the
+code:
+
+.. io-code-block::
+
+   .. input:: /includes/fundamentals/write-operations/WriteOperationsTest.php
+      :language: php
+      :dedent:
+      :start-after: begin model upsert
+      :end-before: end model upsert
+
+   .. output::
+      :language: json
+      :visible: false
+
+      {
+        "_id": "660c...",
+        "performer": "Jon Batiste",
+        "venue": "Radio City Music Hall",
+        "genres": [
+          "R&B",
+          "soul"
+        ],
+        "ticketsSold": 4000,
+        "updated_at": ...
+      }
+
+.. _laravel-modify-documents-arrays:
+
+Update Arrays in a Document
+---------------------------
+
+In this section, you can see examples of the following operations that
+update array values in a MongoDB document:
+
+- Add values to an array
+- Remove values from an array
+- Update the value of an array element
+
+These examples modify the sample document created by the following insert
+operation:
+
+.. literalinclude:: /includes/fundamentals/write-operations/WriteOperationsTest.php
+   :language: php
+   :dedent:
+   :start-after: begin array example document
+   :end-before: end array example document
+
+.. _laravel-modify-documents-add-array-values:
+
+Add Values to an Array Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This section shows how to use the ``push()`` method to add values to an array
+in a MongoDB document. You can pass one or more values to add and set the
+optional parameter ``unique`` to ``true`` to skip adding any values
```

Comment:
I: it doesn't seem accurate to say that setting `unique` to `true` skips adding "any values in the array" - maybe you could add "duplicate"?

S: I also think you can clarify this sentence a little by adding a comma:
```suggestion
in a MongoDB document. You can pass one or more values to add to the array, and you can set the
optional parameter ``unique`` to ``true`` to skip adding any duplicate values
```

Reply:
Good catch! I think I missed that when revising this sentence multiple times.